docs/latest/design/egctl.md

```diff
@@ -0,0 +1,73 @@
+# Introduce egctl
+
+## Motivation
+
+EG should provide a command line tool with following capabilities:
+
+- One-step installation/uninstallation experience
+- Collect configuration from envoy proxy and gateway
+- Update admin options such as changing log level
+- Analyse system configuration to diagnose any issues in envoy gateway
+
+This tool is named `egctl`.
+
+## Syntax
+
+Use the following syntax to run `egctl` commands from your terminal window:
+
+```console
+egctl [command] [entity] [name] [flags]
+```
+
+where `command`, `name`, and `flags` are:
+
+* `command`: Specifies the operation that you want to perform on one or more resources,
+  for example `install`, `uninstall`, `config`, `version`.
+
+* `entity`: Specifies the entity the operation is being performed on such as `envoy-proxy` or `envoy-gateway`.
+
+* `name`: Specifies the name of the specified instance. 
+
+* `flags`: Specifies optional flags. For example, you can use the `-c` or `--config` flags to specify the values for installing.
+
+If you need help, run `kubectl help` from the terminal window.
+
+## Operation
+
+The following table includes short descriptions and the general syntax for all the `egctl` operations:
+
+| Operation   | Syntax                           | Description                                                                 |
+| ----------- | -------------------------------- | --------------------------------------------------------------------------- |
+| `version`   | `egctl version`                  | Prints out build version information.                                       |
+| `install`   | `egctl install -c CUSTOMPROFILE` | Install or reconfigure EG on a cluster.                                     |
+| `uninstall` | `egctl uninstall`                | Uninstall EG from a cluster                                                 |
+| `config`    | `egctl config ENTITY`            | Retrieve information about proxy configuration from envoy proxy and gateway |
```

Comment:
Since `config` supports EG and proxy, s/proxy configuration/configuration/

Reply:
Based on the description of `config`, IMO `get` is a better-suited verb. For example:

- To list all running Envoy proxy instances (pods for the Kube provider):  `egctl get envoy-proxy`.
- To list a specific running Envoy proxy instance (pod for the Kube provider):  `egctl get envoy-proxy <name>`.
- To get the runtime config of a specific Envoy Proxy instance (pod for the Kube provider):  `egctl get envoy-proxy <name> --runtime-config`. This is similar to `127.0.0.1:19000/config_dump`.
- To get the bootstrap config of a specific Envoy Proxy instance (pod for the Kube provider):  `egctl get envoy-proxy <name> --bootstrap-config`.

__Note:__ The `--namespace` flag is not specified so egctl will use the default `envoy-gateway-system` namespace.

Additionally, since egctl should support interacting with multiple EGs, `config` should be designated to configure egctl. For example:

- To display egctl's current config context: `egctl config get-contexts`
- To set a context entry in the egctl config file: `egctl config set-context`
- etc.